Service/SchedulerService.php

```
@@ -0,0 +1,186 @@
+     *
+     * @return ScheduledCommand
+     * @throws \ErrorException
+     */
+    private function getCommand ( ) {
+        if ( $this->command) {
+            return $this->command;
+        }
+        
+        if ( !$this->commandName ){
+            throw new \ErrorException('Missing Command Name.');
+        }
+        
+        $cmd = $this->doctrine->getRepository(ScheduledCommand::class)->findOneBy([
+                    'name' => $this->commandName
+                ]);
+        
+        if ( $cmd instanceof ScheduledCommand) {
+            return $cmd;
+        }
+        
+        throw new \ErrorException('Command Not Found.');
+    }
+
+
+    /**
+     * Schedule command to be executed in the next execution cycle
+     *
+     * @param string $action
+     * @return bool
+     */
+    private function commandAction ( string $action ){
+        try {            
+            $cmd = $this->getCommand();
+
+            switch ( $action ){
+                case 'run': $cmd->setExecuteImmediately(true); break;
+                case 'stop': $cmd->setExecuteImmediately(false); break;
+                case 'disable': $cmd->setDisabled(true); break;
+                case 'enable': $cmd->setDisabled(false); break;
+                case ScheduledCommand::MODE_ONDEMAND: $cmd->setExecutionMode( ScheduledCommand::MODE_ONDEMAND); break;
+                case ScheduledCommand::MODE_AUTO: 
+                    if ( CronExpressionLib::isValidExpression( $cmd->getCronExpression()) ) {
+                        $cmd->setExecutionMode( ScheduledCommand::MODE_AUTO); 
+                    } else {
+                        throw new \InvalidArgumentException('Invalid Cron Expression.');
+                    }
+                    break;
+            }
+            $this->doctrine->getManager()->persist($cmd);
+            $this->doctrine->getManager()->flush($cmd);
+
+            return true;
+        } catch (\Exception $ex) {// TODO: Improve error handling            
```

TODO: Improve error handling
I'd feel good if it was changed to an @FUTURE and some log output was used rather than silently ignoring the exception.